src/expr/src/expr/template.rs

```diff
@@ -28,10 +28,10 @@ use crate::expr::{BoxedExpression, Expression};
 
 macro_rules! gen_eval {
     { ($macro:ident, $macro_row:ident), $ty_name:ident, $OA:ty, $($arg:ident,)* } => {
-        fn eval(&self, data_chunk: &DataChunk) -> $crate::Result<ArrayRef> {
+        async fn eval(&self, data_chunk: &DataChunk) -> $crate::Result<ArrayRef> {
```

Comment:
concretely, for desugaring async_trait, this function would have to return `Pin<Box<dyn Future + Send>>`

See also: https://blog.rust-lang.org/inside-rust/2022/11/17/async-fn-in-trait-nightly.html

Anw, another option, since we are compiling with nightly anw, is to simply avoid `async_trait` and try `#![feature(async_fn_in_trait)]`? Does it work?

Reply:
Manually desugaring seems to be a promising solution. But the async-fn-in-trait feature may not work, because IIRC dynamic dispatch is still in an early stage and not available in nightly.

Reply:
It worked!